File: app/src/main/java/com/druger/aboutwork/fragments/ChangeEmailFragment.java
package com.druger.aboutwork.fragments;


import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import com.druger.aboutwork.R;
import com.druger.aboutwork.activities.LoginActivity;
import com.druger.aboutwork.interfaces.view.ChangeEmailView;
import com.druger.aboutwork.presenters.ChangeEmailPresenter;

import moxy.presenter.InjectPresenter;

import static com.druger.aboutwork.Const.Bundles.EMAIL;

public class ChangeEmailFragment extends BaseSupportFragment implements ChangeEmailView {

    @InjectPresenter
    ChangeEmailPresenter changeEmailPresenter;

    private EditText etEmail;
    private Button btnChangeEmail;
    private String email;

    public static ChangeEmailFragment newInstance(String email) {

        Bundle args = new Bundle();
        args.putString(EMAIL, email);

        ChangeEmailFragment fragment = new ChangeEmailFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        rootView = inflater.inflate(R.layout.fragment_change_email, container, false);
        setupUI();
        setupUX();
        setupToolbar();
        showEmail();
        return rootView;
    }

    private void setupToolbar() {
        mToolbar = bindView(R.id.toolbar);
        setActionBar(mToolbar);
        getActionBar().setTitle(R.string.change_email);
        getActionBar().setDisplayHomeAsUpEnabled(true);
    }

    private void showEmail() {
        email = getArguments().getString(EMAIL);
        if (email != null && email.equals(getString(R.string.add_email))) {
           email = "";
        } else etEmail.setText(email);
    }

    private void setupUX() {
        btnChangeEmail.setOnClickListener(v -> changeEmailPresenter.changeEmail(email, getActivity()));
        etEmail.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                email = s.toString();
            }
        });
    }

    private void setupUI() {
        etEmail = bindView(R.id.etEmail);
        btnChangeEmail = bindView(R.id.btnChangeEmail);
        mProgressBar = bindView(R.id.progressBar);
    }

    @Override
    public void showLoginActivity() {
        startActivity(new Intent(getActivity(), LoginActivity.class));
    }

    @Override
    public void showProgress(boolean show) {
        super.showProgress(show);
        if (show) {
            etEmail.setVisibility(View.INVISIBLE);
            btnChangeEmail.setVisibility(View.INVISIBLE);
        } else {
            etEmail.setVisibility(View.VISIBLE);
            btnChangeEmail.setVisibility(View.VISIBLE);
        }
    }
}

File: app/src/main/java/com/druger/aboutwork/fragments/ChangeEmailFragment.kt
package com.druger.aboutwork.fragments


import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.druger.aboutwork.Const.Bundles.EMAIL
import com.druger.aboutwork.R
import com.druger.aboutwork.activities.LoginActivity
import com.druger.aboutwork.interfaces.view.ChangeEmailView
import com.druger.aboutwork.presenters.ChangeEmailPresenter
import kotlinx.android.synthetic.main.fragment_change_email.*
import kotlinx.android.synthetic.main.toolbar.*
import moxy.presenter.InjectPresenter

class ChangeEmailFragment : BaseSupportFragment(), ChangeEmailView {

    @InjectPresenter
    lateinit var changeEmailPresenter: ChangeEmailPresenter

    private var email: String? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        rootView = inflater.inflate(R.layout.fragment_change_email, container, false)
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupUI()
        setupUX()
        setupToolbar()
        showEmail()
    }

    private fun setupToolbar() {
        mToolbar = toolbar
        setActionBar(mToolbar)
        actionBar.setTitle(R.string.change_email)
        actionBar.setDisplayHomeAsUpEnabled(true)
    }

    private fun showEmail() {
        email = arguments?.getString(EMAIL)
        if (email != null && email == getString(R.string.add_email)) {
            email = ""
        } else
            etEmail.setText(email)
    }

    private fun setupUX() {
        btnChangeEmail.setOnClickListener { email?.let { it -> changeEmailPresenter.changeEmail(it) } }
        etEmail.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}

            override fun afterTextChanged(s: Editable) {
                email = s.toString()
            }
        })
    }

    private fun setupUI() {
        mProgressBar = progressBar
    }

    override fun showLoginActivity() {
        val intent = Intent(activity, LoginActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
    }

    override fun showProgress(show: Boolean) {
        super.showProgress(show)
        if (show) {
            etEmail.visibility = View.INVISIBLE
            btnChangeEmail.visibility = View.INVISIBLE
        } else {
            etEmail.visibility = View.VISIBLE
            btnChangeEmail.visibility = View.VISIBLE
        }
    }

    companion object {

        fun newInstance(email: String): ChangeEmailFragment {

            val args = Bundle()
            args.putString(EMAIL, email)

            val fragment = ChangeEmailFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
